specification/monitor/resource-manager/microsoft.insights/2017-05-01-preview/diagnosticsSettingsCategories_API.json

```diff
@@ -0,0 +1,266 @@
+      "get": {
+        "tags": [
+          "DiagnosticSettingsCategories"
+        ],
+        "operationId": "DiagnosticSettingsCategory_List",
+        "description": "Lists the diagnostic settings categories for the specified resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get more information about diagnostic setting category",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsCategoryResourceCollection"
+            },
+            "examples": {
+              "application/json": {
+                "value": [
+                  {
+                    "id":"/subscriptions/1a66ce04-b633-4a0b-b2bc-a912ec8986a6/resourcegroups/viruela1/providers/microsoft.logic/workflows/viruela6/providers/microsoft.insights/diagnosticSettingsCategories/WorkflowRuntime",
+                    "type":"microsoft.insights/diagnosticSettingsCategories",
+                    "name":"WorkflowRuntime",
+                    "location":"eastus",
+                    "tags":null,
+                    "properties": {
+                      "categoryType":"Logs"
+                    }
+                  },
+                  {
+                    "id":"/subscriptions/1a66ce04-b633-4a0b-b2bc-a912ec8986a6/resourcegroups/viruela1/providers/microsoft.logic/workflows/viruela6/providers/microsoft.insights/diagnosticSettingsCategories/WorkflowMetric",
+                    "type":"microsoft.insights/diagnosticSettingsCategories",
+                    "name":"WorkflowMetric",
+                    "location":"eastus",
+                    "tags":null,
+                    "properties": {
+                      "categoryType":"Metrics"
+                    }
+                  }
+                ]
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting": {
+            "$ref": "./examples/listDiagnosticSettingsCategories.json"
+          }
+        } 
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource name"
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource type"
+        },
+        "location": {
```

Comment:
I don't think there should be location or tags here. Isnt this a proxy resource.

Reply:
It is a proxy resource, but these are populated anyway when the request comes back. Also the resource class is shared by everything. It has been that way for a long time.

I believe we could have another class for proxy resource, but I prefer addressing that in a different release.

Reply:
Can we get an ACK in email with a workitem that this will be addressed in the next release? Apologies, but this is the only way to get the APIs cleaned up across Azure plat. I wont remember this comment in a few months and no one else on your team will :). So it will most likely be missed in the next rev also.

Reply:
Created a task for this